downloader/src/main/kotlin/vcs/Mercurial.kt

```diff
@@ -65,7 +65,7 @@ class Mercurial : VersionControlSystem() {
     override fun getWorkingTree(vcsDirectory: File) = MercurialWorkingTree(vcsDirectory, type)
 
     override fun isApplicableUrlInternal(vcsUrl: String) =
-        ProcessCapture("hg", "identify", vcsUrl).isSuccess
+        MercurialCommand.isInPath() && ProcessCapture("hg", "identify", vcsUrl).isSuccess
```

Comment:
Would it make sense to call `isAvailable()` instead of `MercurialCommand.isInPath()`?

Reply:
The actual problem I try to solve here is that 'isAvailable()` for Mercurial returns *true* even if the `hg` command is not on the path. This is because the available check in `VersionControlSystem` is implemented as
`this !is CommandLineTool || isInPath()`

`Mercurial` is not a `CommandLineTool` (since the command line tool implementation is delegated to `MercurialCommand`), therefore, this check yields always *true*.

I think, this answers all your comments @fviernau, doesn't it?

Reply:
I believe the `isAvailable()` implementation worked out properly until [1] got merged.
So, [1] forgot to update `isAvailable()`. How about fixing up [1] to address the problem?

[1] https://github.com/oss-review-toolkit/ort/commit/55318202ea53c66a213f0c8fc5bd2fa7ecd440cb

Reply:
I like this, but this would not be sufficient to do the check whether the command is in the path.

How about having a function `getAssociatedCommandLineTool(): CommandLineTool?` or similar in `VersionControlSystem` that returns the tool if there is one? Then all the properties of the returned tool object would be directly available.

Reply:
Another idea: How about simply making `VersionControlSystem.isAvailable()` open and overriding it in `Mercurial` with `= MercurialCommand.isInPath()`? The same should then be done for `Cvs`, BTW.

Reply:
I thought about this, too. One advantage of the solution proposed by @fviernau would be that it could be implemented in `VersionControlSystem` centrally without having to implement the logic in the concrete implementations.

Reply:
I have updated the PR to implement the proposal. @sschuberth, @fviernau could you please have a look?